gdb-xtensa-esp32-elf/build.sh

```diff
@@ -1,18 +1,49 @@
-#!/bin/bash
+#!/bin/usr/env bash
 
-# Copy the ESP32 "overlay" files:
-cp -av $SRC_DIR/crosstool-NG-esp32/overlays/xtensa_esp32/binutils/. $SRC_DIR/binutils-gdb-esp32-src/
-cp -av $SRC_DIR/crosstool-NG-esp32/overlays/xtensa_esp32/gdb/. $SRC_DIR/binutils-gdb-esp32-src/
+###
+# Note
+#
+# This script is partially based on:
+# https://github.com/espressif/binutils-gdb/blob/esp-gdb-v11.1_20220318/build_esp_gdb.sh
+###
 
-export TARGET=xtensa-esp32-elf
+export TARGET=xtensa-esp-elf
```

Comment:
nb this does move the final installation path, but seemed like a fairly harmless change

Reply:
As long as we update the code in all the spots, I'm fine with it of course!

Reply:
this is what was done in https://github.com/espressif/binutils-gdb/blob/1ddf340cfa5313e782e852113e94eaa53b154afd/build_esp_gdb.sh#L89 , but it turns out it doesn't matter, so backing it out!

Reply:
ohhh right. this is because the actual gdb binary is now called `xtensa-esp-gdb`, and the soc-specific variants are named thus:

```bash
❯ ll ${CONDA_PREFIX}/bin/xtensa-esp*elf-gdb
lrwxrwxrwx 1 noah noah 42 Aug 23 11:44 /home/noah/mambaforge/envs/memfault/bin/xtensa-esp32-elf-gdb -> ../xtensa-esp-elf/bin/xtensa-esp32-elf-gdb
lrwxrwxrwx 1 noah noah 44 Aug 23 11:44 /home/noah/mambaforge/envs/memfault/bin/xtensa-esp32s2-elf-gdb -> ../xtensa-esp-elf/bin/xtensa-esp32s2-elf-gdb
lrwxrwxrwx 1 noah noah 44 Aug 23 11:44 /home/noah/mambaforge/envs/memfault/bin/xtensa-esp32s3-elf-gdb -> ../xtensa-esp-elf/bin/xtensa-esp32s3-elf-gdb
lrwxrwxrwx 1 noah noah 40 Aug 23 11:44 /home/noah/mambaforge/envs/memfault/bin/xtensa-esp-elf-gdb -> ../xtensa-esp-elf/bin/xtensa-esp-elf-gdb
```

Which are wrapper scripts:

```bash
❯ file -L ${CONDA_PREFIX}/bin/xtensa-esp*elf-gdb
/home/noah/mambaforge/envs/memfault/bin/xtensa-esp32-elf-gdb:   Bourne-Again shell script, ASCII text executable
/home/noah/mambaforge/envs/memfault/bin/xtensa-esp32s2-elf-gdb: Bourne-Again shell script, ASCII text executable
/home/noah/mambaforge/envs/memfault/bin/xtensa-esp32s3-elf-gdb: Bourne-Again shell script, ASCII text executable
/home/noah/mambaforge/envs/memfault/bin/xtensa-esp-elf-gdb:     ELF 64-bit LSB pie executable, x86-64, version 1 (GNU/Linux), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.18, with debug_info, not stripped
```